### PR TITLE
rc_update: cleanup parameter usage

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

File: src/modules/rc_update/rc_update.cpp [hunk 1]
```diff
@@ -78,8 +106,46 @@ RCUpdate::init()
...
-	update_parameters(_parameter_handles, _parameters);
+	if (_param_rc_flt_smp_rate.get() < 1.0f) {
+		_param_rc_flt_smp_rate.set(1.0f);
+		_param_rc_flt_smp_rate.commit_no_notification();
+	}
+
+	// make sure the filter is in its stable region -> fc < fs/2
+	const float flt_cutoff_min = _param_rc_flt_smp_rate.get() / 2.0f - 1.0f;
+
+	if (_param_rc_flt_cutoff.get() < flt_cutoff_min) {
```
Comment (bkueng) [inline src/modules/rc_update/rc_update.cpp]: It's an upper bound:
```suggestion
	if (_param_rc_flt_cutoff.get() > flt_cutoff_max) {
```

File: src/modules/rc_update/rc_update.cpp [hunk 2]
```diff
@@ -354,7 +415,10 @@ RCUpdate::Run()
 				_rc.channels[i] = 0.0f;
 			}
 
-			_rc.channels[i] *= _parameters.rev[i];
+			if (_parameters.rev[i]) {
+				_rc.channels[i] = -1.0f * _rc.channels[i];
```
Comment (bkueng) [inline src/modules/rc_update/rc_update.cpp]:
```suggestion
				_rc.channels[i] = -_rc.channels[i];
```

File: src/modules/rc_update/parameters.cpp [hunk 3]
```diff
@@ -1,325 +0,0 @@
...
-		param_get(parameter_handles.min[i], &(parameters.min[i]));
-		param_get(parameter_handles.trim[i], &(parameters.trim[i]));
-		param_get(parameter_handles.max[i], &(parameters.max[i]));
-		param_get(parameter_handles.rev[i], &(parameters.rev[i]));
-		param_get(parameter_handles.dz[i], &(parameters.dz[i]));
-
-		tmpScaleFactor = (1.0f / ((parameters.max[i] - parameters.min[i]) / 2.0f) * parameters.rev[i]);
-		tmpRevFactor = tmpScaleFactor * parameters.rev[i];
-
-		/* handle blowup in the scaling factor calculation */
```
Comment (bkueng) [inline src/modules/rc_update/parameters.cpp]: You did not want to keep that check?

Comment (dagar) [reply src/modules/rc_update/parameters.cpp]: I'm not opposed to keeping it, but not in this form printing an error message that's likely never going to be seen by a user. The scaling_factor it clears isn't even used anymore.

File: src/modules/rc_update/rc_update.cpp (hunk 1 above)
Comment (dagar) [reply src/modules/rc_update/rc_update.cpp]: Oops, nice catch.

**Comment by dagar** [comment]
@PX4/testflights on any vehicle could you please try fully (re)calibrating RC and verify everything still works? If it checks out on the bench then please test fly and provide a log that can be compared to master. We can use that to compare calibration results before and after.

**Comment by Junkim3DR** [comment]
### Tested on NXP FMUK66 v3

**Modes Tested**
Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

**Procedure**
- Calibrate RC before flying.
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL and see landing behavior.

**Notes**
No issues noted, good flight in general.

**Log** 
- PR 13672
  - https://review.px4.io/plot_app?log=9dfccd8e-2765-408d-a88a-aac52fd82934
- Master
  - https://review.px4.io/plot_app?log=58aba96b-b0bd-4f8d-98dd-5b8052b6e790





### Tested on Pixhawk 4 v5

**Modes Tested**
Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

**Procedure**
- Calibrate RC before flying.
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL and see landing behavior.

**Notes**
No issues noted, good flight in general.

**Log** 
- PR 13672
  - https://review.px4.io/plot_app?log=a268a939-2dae-4690-96fd-4675f6eeb182
- Master
  - https://review.px4.io/plot_app?log=4e68716c-4c3c-43d4-b4d8-ecc00125d9cb

**Comment by jorge789** [comment]
### Tested on PixRacer V4:
**Modes Tested**
Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

**Procedure**

Calibrate RC before flying.
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL and see landing behavior.
**Notes**
No issues noted, good flight in general.

**Log**

PR 13672
https://review.px4.io/plot_app?log=e6f4466b-2982-4ce9-ba50-0cc38ff76e5f

Master
https://review.px4.io/plot_app?log=56c74d70-5225-47a1-99e4-4ba2aaaacdbf

**Pull request merged**